src/main/fc/cleanflight_fc.c

```diff
@@ -750,40 +769,59 @@ void taskMainPidLoop(void)
     if (motorControlEnable) {
         writeMotors();
     }
+    if (debugMode == DEBUG_PIDLOOP) {debug[3] = micros() - startTime;}
+}
 
-#ifdef USE_SDCARD
-        afatfs_poll();
-#endif
 
-#ifdef BLACKBOX
-    if (!cliMode && feature(FEATURE_BLACKBOX)) {
-        handleBlackbox();
+
+
+uint8_t setPidUpdateCountDown(void) {
+    if (gyroConfig()->gyro_soft_lpf_hz) {
+        return imuConfig()->pid_process_denom - 1;
+    } else {
+        return 1;
     }
-#endif
 }
 
 // Function for loop trigger
-void taskMainPidLoopChecker(void) {
-    // getTaskDeltaTime() returns delta time freezed at the moment of entering the scheduler. currentTime is freezed at the very same point.
-    // To make busy-waiting timeout work we need to account for time spent within busy-waiting loop
-    uint32_t currentDeltaTime = getTaskDeltaTime(TASK_SELF);
-
-    if (imuConfig()->gyroSync) {
-        while (1) {
-            if (gyroSyncCheckUpdate() || ((currentDeltaTime + (micros() - currentTime)) >= (targetLooptime + GYRO_WATCHDOG_DELAY))) {
-                break;
-            }
-        }
+void taskMainPidLoopCheck(void)
+{
+    static bool runTaskMainSubprocesses;
```

Comment:
agreed, there's a few other cases.  dT in pid.c was another one.  IMHO all static variables should be initialized for clarity.